### PR TITLE
Langpicker

### PR DESCRIPTION
**Pull request by toledoal**
Adding the language picker for Docs. 
I can successfully switch languages but I'm not able to get any translations for the docs. Maybe we currently don't have any? 
Any tips on where should I look for to verify? Thanks. 

image

### CODE REVIEW

**Comment by toledoal** [comment]
Linting errors! Will update

File: pxtlib/docsrender.ts [hunk 1]
```diff
@@ -230,6 +230,15 @@ namespace pxt.docs {
             return injectHtml(templ, mparams, ["ITEMS"])
         }
 
+        let availableLocales = theme.selectLanguage && theme.availableLocales && theme.availableLocales.length ?
+        theme.availableLocales : ["en"]
+
+        params["selectLang"] = lf("Select Language");
+        params["crowdin"] = theme.crowdinProject ?
+            `<p><br/><br/><a href="https://crowdin.com/project/${theme.crowdinProject}"
+            target="_blank" rel="noopener noreferrer" aria-label=${lf("Help us translate")}>${lf("Help us translate")}</a>
```
Comment (pelikhan) [inline pxtlib/docsrender.ts]: potential XSS injection via translated strings.

File: docfiles/docs.js [hunk 2]
```diff
@@ -259,6 +260,72 @@ function renderSnippets() {
...
+
+function setupLangPicker() {
+
+    let availableLocales = pxt.appTarget.appTheme.availableLocales;
+    let locales;
+    let initialLang = pxt.Util.normalizeLanguageCode(getCookieLang())[0];
+    if (pxt.appTarget.appTheme && availableLocales) {
+
+        locales = availableLocales.map(e => languageOption(e)).join("\n");
+        document.querySelector("#availablelocales").innerHTML = locales;
```
Comment (riknoll) [inline docfiles/docs.js]: Don't use innerHTML, if you need to inflate HTML we usually use `document.createElement()` directly and `el.textcontent = "whatever"`

**Comment by pelikhan** [comment]
Requires backend deployment.

File: pxtlib/docsrender.ts [hunk 3]
```diff
@@ -230,6 +230,15 @@ namespace pxt.docs {
             return injectHtml(templ, mparams, ["ITEMS"])
         }
 
+        let availableLocales = theme.selectLanguage && theme.availableLocales && theme.availableLocales.length ?
+        theme.availableLocales : ["en"]
+
+        params["selectLang"] = lf("Select Language");
```
Comment (pelikhan) [inline pxtlib/docsrender.ts]: can we move this to the html page?

File: pxtlib/docsrender.ts [hunk 4]
```diff
@@ -230,6 +230,15 @@ namespace pxt.docs {
...
+        params["crowdin"] = theme.crowdinProject ?
```
Comment (pelikhan) [inline pxtlib/docsrender.ts]: we should not send people direclty to crowdin. Instead send them to ``/translate``.

File: docfiles/docs.js (hunk 2 above)
Comment (toledoal) [reply docfiles/docs.js]: good practice, thanks

**Comment by toledoal** [comment]
I still don't know why lf() doesn't work in docs... any guidance? Maybe there is something that needs to be set?

File: pxtlib/docsrender.ts [hunk 5]
```diff
@@ -230,6 +230,8 @@ namespace pxt.docs {
             return injectHtml(templ, mparams, ["ITEMS"])
         }
 
+        params["selectLang"] = lf("Select Language");
```
Comment (pelikhan) [inline pxtlib/docsrender.ts]: do you really need these changes? Any modification of docsrender.ts require a cloud update.

Comment (toledoal) [reply pxtlib/docsrender.ts]: Today I'm going to focus on trying to remove any of my changes that need cloud updates and replace them. I hope I can make this work in all the PR I have open. :)

File: docfiles/docs.js [hunk 6]
```diff
@@ -259,6 +260,93 @@ function renderSnippets() {
...
+   
+        document.querySelector("#langpicker").onclick = function() { 
+            $('.ui.modal').modal('show'); 
+        };
+
+        document.querySelector(".closeIcon").onclick = function() { 
+            $('.ui.modal').modal('hide'); 
+        };
+
+        let langOptions = document.querySelectorAll(".langoption");
```
Comment (pelikhan) [inline docfiles/docs.js]: test in IE11

Comment (toledoal) [reply docfiles/docs.js]: Sure

Comment (pelikhan) [reply docfiles/docs.js]: Not sure “let” is supported over there

Comment (toledoal) [reply docfiles/docs.js]: Correct, those are not supported. Will change across the file. Also will verify anonymous functions are not arrow functions.

**Comment by pelikhan** [comment]
Looks good to me.
Merge https://github.com/microsoft/pxt/pull/5313 before.

File: docfiles/footer.html [hunk 7]
```diff
@@ -22,6 +22,9 @@
             <a class="item" href="https://makecode.com/privacy" target="_blank" rel="noopener">Privacy &amp; Cookies</a>
             <a class="item" href="https://makecode.com/termsofuse" target="_blank" rel="noopener"> Terms Of Use</a>
             <a class="item" href="https://makecode.com/trademarks" target="_blank" rel="noopener">Trademarks</a>
+            <a id="langpicker" class="ui item" rel="noopener">
```
Comment (jwunderl) [inline docfiles/footer.html]:
```suggestion
            <a id="langpicker" class="ui item" rel="noopener" tabindex="0">
```

without an ``href`` this isn't in tab order by default

File: docfiles/docs.js [hunk 8]
```diff
@@ -259,6 +260,93 @@ function renderSnippets() {
...
+            $('.ui.modal').modal('show'); 
+        };
+
+        document.querySelector(".closeIcon").onclick = function() { 
+            $('.ui.modal').modal('hide'); 
+        };
+
+        var langOptions = document.querySelectorAll(".langoption");
+
+        for(var x=0; x<langOptions.length; x++) {
```
Comment (jwunderl) [inline docfiles/docs.js]:
```suggestion
        for(var x = 0; x < langOptions.length; x++) {
```

nit: spacing

**Comment by jwunderl** [comment]
superceded by https://github.com/microsoft/pxt/pull/6064

**Pull request closed without merging**